eventloop/eventloop.go

```diff
@@ -56,8 +73,13 @@ func (e *EventLoop) awaitAll() {
 	n := len(e.promiseQueue)
 	for i := n - 1; i >= 0; i-- {
 		p := e.promiseQueue[i]
-		if p.handler {
-			<-p.done
+		select {
+		case <-p.errChan:
+			continue
+		default:
+			if p.handler {
+				<-p.done
+			}
```

Comment:
we should only wait on done signal in the even-loop

Reply:
Fixed.